python/pipeline/pupil.py

```diff
@@ -596,27 +596,75 @@ class Tracking(dj.Computed):
+        """
+
     class ManualTracking(dj.Part):
         definition="""
         -> master
         frame_id                    : int                   # frame id (starting from 0)
         ---
         contour=NULL                : longblob              # eye contour relative to ROI
         """
-        def make(self, key):
-            print("""
-            This method is deprecated!
-            Simply re-inserting data that were manually tracked before (ManuallyTrackedContours)!
-            If data that matches with key does not exist, it will not be stored!
-            """)
-
-            if (ManuallyTrackedContours() & key).fetch1() is not None:
-                for frame_id in range((ManuallyTrackedContours.Frame & key).fetch('frame_id').max()):
-                    ckey = (ManuallyTrackedContours.Frame & dict(key, frame_id=frame_id)).fetch1()
-                    self.insert1(dict(ckey, tracking_method= key['tracking_method']))
+        def make(self, key, backup_file=None):
+            
+            # key does exist in ManuallyTrackedContours (i.e. we tracked before)
+            if len(ManuallyTrackedContours() & key):
+
+                print("""
+                The given key has been tracked manually before (from ManuallyTrackedContours)! 
+                Simply re-inserting previously tracked data here!
+                """)
+                for frame_id in range(len(ManuallyTrackedContours.Frame & key)):
+                    # copy Frame info
+                    frame_key = (ManuallyTrackedContours.Frame & dict(key, frame_id=frame_id)).fetch1()
+                    self.insert1(dict(frame_key, tracking_method= key['tracking_method']))
+                    # copy Parameter info
+                    param_key = (ManuallyTrackedContours.Parameter & dict(key, frame_id=frame_id)).fetch1()
+                    min_lambda = (ManuallyTrackedContours & key).fetch1('min_lambda')
+                    Tracking.ManualTrackingParameter.insert1(
+                        dict(param_key, min_lambda=min_lambda, tracking_method= key['tracking_method']))
+            
+            # key does not exist in ManuallyTrackedContours, hence need to trace manually
             else:
-                print("Given key does not exist in ManuallyTrackedContours table!")
-                print("Either manually track by populating ManuallyTrackedContours or use deeplabcut method!")
+                print("Manually Tracking!")
+            
+                if backup_file is None:
+                    avi_path = (Eye() & key).get_video_path()
+                    tracker = ManualTracker(avi_path)
+                    tracker.backup_file = '/tmp/tracker_state{animal_id}-{session}-{scan_idx}.pkl'.format(
+                        **key)
+                else:
+                    tracker = ManualTracker.from_backup(backup_file)
+
+                try:
+                    tracker.run()
```

Comment:
you also should take this out of ManuallyTrackedContours so if somebody wants to populate ManualllyTrackedcontours it should say it's deprecated and ask it to populate Tracking with tracking_method manual.

Reply:
Do you mean, if I run `ManuallyTrackedContours.populate` it should prompt deprecation warning and does not do anything?